code/modules/mob/say.dm

```diff
@@ -25,7 +27,12 @@
 	*/
 
 	set_typing_indicator(0)
-	usr.say(message)
+	if(world.time > say_cooldown)
+		usr.say(message)
+		say_cooldown = world.time + 10 // We said something so take some breathing room.
+	else
+		to_chat(src, "<span class='notice'>Take a breath before speaking again!</span>") // Notify the user they've said things too fast.

```

Comment:
Spammers spamming themselves can still lag the server by processing requests. Ideally you should just have `if(world.time > say_cooldown) return` above `usr.say(message)`.

Reply:
Addressed